pkg/kubelet/kubelet.go

```diff
@@ -2107,19 +2033,17 @@ func getPodReadyCondition(spec *api.PodSpec, info api.PodInfo) []api.PodConditio
 }
 
 // GetPodStatus returns information from Docker about the containers in a pod
-func (kl *Kubelet) GetPodStatus(podFullName string, uid types.UID) (api.PodStatus, error) {
-	uid = kl.translatePodUID(uid)
-
+func (kl *Kubelet) GetPodStatus(podFullName string) (api.PodStatus, error) {
 	// Check to see if we have a cached version of the status.
-	cachedPodStatus, found := kl.getPodStatusFromCache(podFullName)
+	cachedPodStatus, found := kl.statusManager.GetPodStatus(podFullName)
 	if found {
 		glog.V(3).Infof("Returning cached status for %q", podFullName)
 		return cachedPodStatus, nil
 	}
-	return kl.generatePodStatus(podFullName, uid)
+	return kl.generatePodStatus(podFullName)
```

Comment:
Kubelet failed to retrieve cached PodStatus from statusManager, regenerate it from scratch. Shouldn't this update the cached one in statusManager too?